README.md

```diff
@@ -153,6 +153,13 @@ usage is clear you can use `U`, `V` etc. for any following generic.
 If the usage is not obvious, you should use a more descriptive name. The same naming rules as for
 classes will apply then.
 
+### Prefixing
+
+#### Types and Interfaces
```

Comment:
Can you add some do/don't examples? :)

Reply:
Hopefully it's more clear now.

Reply:
I meant more like this :D

Do ✅ 
```ts
// using T, U, V, ... when they are generic
// and P for key iteration
type Partial<T> = { [P in keyof T]?: T[P]; };
```
```ts
// or K/V for keys and values
interface Map<K, V> {
}
```
```ts
// or write them out
interface Map<Key, Value> {
}
```

Dont 🚫 
```ts
// prefix with T
interface Map<TKey, TValue> {
}
```

Reply:
We shouldn't use `K` and `V` as abbreviations for `Key` and `Value`, that is not clear enough. So I don't think that is a good example. We should avoid using abbreviations.

Reply:
I disagree, even the `Map` type in TS is defined using the abbreviated generic names. Same for using `P` as a generic for props. We shouldn't hide all abbreviations and people should learn the standard ones used in the industry.

Reply:
I agree with @psimk here `<K, V>` are common types. If wouldn't allow this we would even have to write down `Type` instead of `T`.

Reply:
Exactly, I would say to prefer the 1-letter conventions that exist for specific cases (T, U, V... K, V.. P). Everyone that works with types know what they mean, same as `i` and `j` for iterators.

Even more so, in certain libraries you can start domain-specific conventions, like jsonapi-client uses `R` for any resource generic type, and `E` for any endpoint generic type. This is super clear when using it further down in the type.

Even more so, when using full names like `Key`, my brain thinks there is an existing type somewhere else outside of the "generic scope". So there is a clear split between "internal generics" and "outside types". Just something that's a nice benefit.

Reply:
Using full/verbose names for type params will get messy, I tried this with jsonapi-client (albeit a perhaps overly complex example for general usage) but that got out of hand real fast.

Reply:
One thing that is a a bit confusion is that the title says `Types and Interfaces` but the example is for `Generics`.

Maybe we should create a separate chapter for Generics and add another example for the Types and Interfaces

Reply:
I suggest the following:

1. Move the do's and don'ts from @ThaNarie & @mmnathan examples to https://github.com/mediamonks/frontend-coding-standards#typescript-generics
2. Remove the `#### Types and Interfaces` chapter
3. Add an example of `TData`, to the don'ts.

Reply:
> Remove the #### Types and Interfaces chapter

This one is still relevant, but needs different examples, specifically:

```
type XX = ...
interface XX { ... }
```

And perhaps links to:
* that article with a table that shows the differences between types and interfaces
* that article that explains why interfaces are more performant than types

Reply:
@ThaNarie Updated documentation. I didn't put a clear stamp on to prefering interfaces over types. Maybe that's another point of discussion ;).